services/app-api/src/authn/cognitoAuthn.ts

```diff
@@ -126,45 +144,153 @@ export function userTypeFromAttributes(attributes: {
             role: 'CMS_USER',
             email: attributes.email,
             name: fullName,
+            givenName: attributes.given_name,
+            familyName: attributes.family_name,
+            euaID: euaID,
         })
     }
 
     return err(new Error('Unsupported user role:  ' + roleAttribute))
 }
 
+// This converts from the Prisma `User` to the app's `UserType`
+export function userTypeFromUser(user: User): Result<UserType, Error> {
+    if (user.role === 'ADMIN_USER') {
+        return ok({
+            role: 'ADMIN_USER',
+            email: user.email,
+            givenName: user.givenName,
+            familyName: user.familyName,
+            name: user.givenName + ' ' + user.familyName,
+            euaID: user.euaID,
+        })
+    }
+
+    if (user.role === 'CMS_USER') {
+        return ok({
+            role: 'CMS_USER',
+            email: user.email,
+            givenName: user.givenName,
+            familyName: user.familyName,
+            name: user.givenName + ' ' + user.familyName,
+            euaID: user.euaID,
+        })
+    }
+
+    if (user.role === 'STATE_USER') {
+        return ok({
+            role: 'STATE_USER',
+            email: user.email,
+            givenName: user.givenName,
+            familyName: user.familyName,
+            name: user.givenName + ' ' + user.familyName,
+            euaID: user.euaID,
```

Comment:
Are we certain state users have an eua? I thought EUA was only something CMS users would have since they use EUA sso login when they access IDM. State users just go directly through IDM. 

Also, if its true that only CMS users have EUA, I start to wonder do we really even need to track EUA in our app ? What is the value of that?